### ISSUE TITLE
error adding complex[double] to double

### ISSUE THREAD

**Issue by axil** [cython/cython#1643]
According to `libcpp/complex.pxd` adding `T` to `complex[T]` is supported:

        complex[T] operator+(complex[T]&, T&)
        complex[T] operator+(T&, complex[T]&)

But it doesn't work:

a.pyx:

    # distutils: language = c++
    
    cimport libcpp.complex
    
    def f():
        libcpp.complex.complex[double](1,2) + libcpp.complex.complex[double](2,3) # ok
        libcpp.complex.complex[double](1,2) + 5. # Cannot assign type 'double' to 'complex[double]' 
        5. + libcpp.complex.complex[double](1,2) # Invalid operand types for '+' (double; complex[double])

setup.pyx:

    from distutils.core import setup
    from Cython.Build import cythonize
    
    setup(
        name = "demo",
        ext_modules = cythonize('a.pyx'),
    )

Àny idea how to fix it?

Moving declaration

    complex[T] operator+(complex[T]&, T&)

out of `cppclass` and changing it to

    complex[T] operator+[T](complex[T]&, T&)

looks more legitimate but still does not work.

**Comment by axil** [comment]
The commit above fixes the issue. After applying it, the operators can be used this way:

a.pyx:
```
# distutils: language = c++

cimport libcpp.complex                     # imports classes and functions
from libcpp.complex cimport *          # imports the operators

def f():
    libcpp.complex.complex[double](1,2) + libcpp.complex.complex[double](2,3) # ok
    libcpp.complex.complex[double](1,2) + 5. # ok
    5. + libcpp.complex.complex[double](1,2) # ok
```

(thanks DavidW from stackoverflow for lending a hand :) ).

**Comment by raffienficiaud** [comment]
I believe this is fixed by #6040

**Comment by matusvalo** [comment]
> I believe this is fixed by #6040

I confirm.
